source/lac/petsc_solver.cc

```diff
@@ -104,6 +92,18 @@ namespace PETScWrappers
         ierr = KSPSetPC (solver_data->ksp, preconditioner.get_pc());
         AssertThrow (ierr == 0, ExcPETScError(ierr));
 
+        // setting the preconditioner overwrites the used matrices.
+        // hence, we need to set the matrices after the preconditioner.
+        // the last argument is irrelevant here,
```

Comment:
would you be willing to move it there? I think it reads better that way, although it's clear that it applies to the first part of in-else-endif.